### PR TITLE
[Reference Architecture] Principles & Requirements

### PR DESCRIPTION
**Pull request by rabi-abdel**

### CODE REVIEW

**Comment by rabi-abdel** [comment]
Added Content provided ny @pgoyal01 (Issue: https://github.com/cntt-n/CNTT/issues/171)

**Comment by walterkozlowski** [comment]
@pgoyal01 I have made changes you proposed and some more

**Comment by walterkozlowski** [comment]
@pgoyal01 Pankaj, I made changes you suggested and did few more updates.  Can you have a look at this little section and let me know how it looks by now? Thx.

**Comment by pgoyal01** [comment]
@walterkozlowski Looks good. Thanks.

File: doc/ref_arch/README.md [hunk 1]
```diff
@@ -2,15 +2,38 @@
...
+
+There are a number of key architectural principles that apply to all reference architectures produced by CNTT. These principles are enumerated below. They are meant to be general and at a high level, and limited in number. Some of the chapters of this document will include more specific principles to provide the implementation guidelines for a particular function or a specific component. Note that the Architectural Principles discussed in this Chapter follow the over-arching principles provided in [CNTT Reference Model:Introduction:Principles](https://cntt-n.github.io/CNTT/doc/ref_model/chapters/chapter01.html#1.3)
+
+
+We need to distinguish between architectural principles and architectural requirements elaborated later in this document. The principles are here to guide our architectural thinking, while requirements should be understood as a check list used to gauge a level of compliance of a NFVI implementation to the CNTT reference architecture.
+
+1. **Open source preference:** To ensure that suppliers’ and operators’ investment have a tangible pathway towards a standard and production ready NFVI solution portfolio.
```
Comment (CsatariGergely) [inline doc/ref_arch/README.md]: It should be somehow highlited, that the Reference Architectures should build on on technology already available in open source projects.
Maybe we could even have a principle to use only technologies available.

File: doc/ref_arch/README.md [hunk 2]
```diff
@@ -2,15 +2,38 @@
...
 
-<a name="1"></a>
+<a name="available-ra"></a>
 ## Available Reference Architectures
 * [Openstack Based](openstack)
 
-<a name="2"></a>
+<a name="principles"></a>
 ## Principles
-Those principles apply to all reference architecture produced by CNTT.
```
Comment (CsatariGergely) [inline doc/ref_arch/README.md]: nit: Reference Architecture

File: doc/ref_arch/README.md [hunk 3]
```diff
@@ -2,15 +2,38 @@
...
+
+
+We need to distinguish between architectural principles and architectural requirements elaborated later in this document. The principles are here to guide our architectural thinking, while requirements should be understood as a check list used to gauge a level of compliance of a NFVI implementation to the CNTT reference architecture.
+
+1. **Open source preference:** To ensure that suppliers’ and operators’ investment have a tangible pathway towards a standard and production ready NFVI solution portfolio. 
+
+1. **Open APIs:** To enable interoperability and component substitution, and minimize integration efforts.
```
Comment (CsatariGergely) [inline doc/ref_arch/README.md]: Open API definition itself is not clear. Does this mean that all API-s should have an [OpenAPI](https://www.openapis.org/) definition or it means that all API-s should be provided by open source projects (what is a consequence of the previous point)?

File: doc/ref_arch/README.md [hunk 4]
```diff
@@ -2,15 +2,38 @@
...
+
+1. **Seperation of concerns:** To promote lifecycle indpendence of different architectural layers and modules (e.g. disagregation of software from hardware).
+
+1. **Automated lifecycle management:** To minimize costs of the end-to-end lifecycle, maintenance downtime (target zero downtime), avoid errors and discrepancies resulting from manual processes.
```
Comment (CsatariGergely) [inline doc/ref_arch/README.md]: This should be scoped only to the infrastructure itself.

File: doc/ref_arch/README.md [hunk 5]
```diff
@@ -2,15 +2,38 @@
...
+
+1. **Automated scalability:** To minimize costs and operational impacts through automated policy-driven scaling of workloads.
+
+1. **Automated closed loop assurance:** To minimize operational costs and simplify operations.
```
Comment (CsatariGergely) [inline doc/ref_arch/README.md]: For me this is unclear. What does it mean for the NFVI?

File: doc/ref_arch/openstack/chapters/chapter01.md [hunk 6]
```diff
@@ -27,27 +27,12 @@
...
-    1. Elastic
-    1. Secure
-    1. Low Latency
-    1. Resource and Operational Efficiency
-    1. E2E Lifecycle Automation (Deployment, Operations, & Maintenance)
-    1. High-Availability
-1. Prioritize incorporation of open source components 
-    1. Design architectures to established open standards as much as possible
-1. Architectures will evolve over time 
+Open Stack Reference Architecture must obey to the following set of principles:
```
Comment (CsatariGergely) [inline doc/ref_arch/openstack/chapters/chapter01.md]: nit: OpenStack

File: doc/ref_arch/README.md [hunk 7]
```diff
@@ -2,15 +2,38 @@
...
+
+
+We need to distinguish between architectural principles and architectural requirements elaborated later in this document. The principles are here to guide our architectural thinking, while requirements should be understood as a check list used to gauge a level of compliance of a NFVI implementation to the CNTT reference architecture.
+
+1. **Open source preference:** To ensure that suppliers’ and operators’ investment have a tangible pathway towards a standard and production ready NFVI solution portfolio. 
+
+1. **Open APIs:** To enable interoperability and component substitution, and minimize integration efforts.
+
+1. **Seperation of concerns:** To promote lifecycle indpendence of different architectural layers and modules (e.g. disagregation of software from hardware).
```
Comment (CsatariGergely) [inline doc/ref_arch/README.md]: nit: Seperation -> Separation
nit: indpendence -> independence

File: doc/ref_arch/README.md (hunk 1 above)
Comment (walterkozlowski) [reply doc/ref_arch/README.md]: Added a clause on this

File: doc/ref_arch/README.md (hunk 2 above)
Comment (walterkozlowski) [reply doc/ref_arch/README.md]: fixed

File: doc/ref_arch/README.md (hunk 3 above)
Comment (walterkozlowski) [reply doc/ref_arch/README.md]: clarified

File: doc/ref_arch/README.md (hunk 4 above)
Comment (walterkozlowski) [reply doc/ref_arch/README.md]: By default all principles apply to the CNTT scope as defined in CNTT RM

File: doc/ref_arch/README.md (hunk 7 above)
Comment (walterkozlowski) [reply doc/ref_arch/README.md]: fixed, thank you

File: doc/ref_arch/README.md (hunk 5 above)
Comment (walterkozlowski) [reply doc/ref_arch/README.md]: It looked quite clear to me but I have added some clarifications for the purists :)

File: doc/ref_arch/README.md (hunk 1 above)
Comment (CsatariGergely) [reply doc/ref_arch/README.md]: Looks okay, thanks.

File: doc/ref_arch/README.md (hunk 3 above)
Comment (CsatariGergely) [reply doc/ref_arch/README.md]: This is clear now.

File: doc/ref_arch/README.md (hunk 4 above)
Comment (CsatariGergely) [reply doc/ref_arch/README.md]: Okay, scope is clear from [Chapter 1 of the RM](https://github.com/cntt-n/CNTT/blob/master/doc/ref_model/chapters/chapter01.md#15scope), but there are usually misunderstantings about the scope. Okay to keep it like this now, but maybe we will need to be more clear if there are lost of misreads of this.

File: doc/ref_arch/README.md (hunk 5 above)
Comment (CsatariGergely) [reply doc/ref_arch/README.md]: Looks more clear now.

**Pull request merged**